### ISSUE TITLE
`__IMPOSSIBLE__` when tactic is used on record fields that are functions

### ISSUE THREAD

**Issue by Trebor-Huang** [agda/agda#5712]
MWE:

```agda
open import Agda.Builtin.Reflection
open import Agda.Builtin.Unit
module MWE where

nothing : Term -> TC ⊤
nothing hole = returnTC _

record Foo : Set1 where
    field
        A : Set
        @(tactic nothing) foo : A -> A

F : Foo
Foo.A F = ⊤
Foo.foo F n = _
```

This produces:

```
An internal error has occurred. Please report this as a bug.
Location of the error: __IMPOSSIBLE__, called at src/full/Agda/TypeChecking/Coverage.hs:521:22 in Agd-2.6.2.1-2f8a192b:Agda.TypeChecking.Coverage
```

It only occurs when `foo` is a function, and the definition of `F` includes that field. (When the last line in the code is left out, it runs fine because Agda can infer a value of `⊤` even when the tactic is doing nothing.)

**Comment by andreasabel** [comment]
This passes without error on 2.6.1. Running bisect:
```
agda-bisect --good v2.6.1 --bad v2.6.2 -w ghc-8.8.4 --with-cabal=cabal-3.4 Issue5712.agda
```

**Comment by andreasabel** [comment]
Error here:https://github.com/agda/agda/blob/9428b1db46b28a4a01706a9347091b09b7136388/src/full/Agda/TypeChecking/Coverage.hs#L480
This is a regression in 2.6.2 introduced by me by the fix of #5358:
0cdd208baf18d7d8f9c5fe24a766a74c357b0151 is the first bad commit
Date:   Fri May 7 21:51:20 2021 +0200
[ fixed #5358 ] don't expand clauses with tactics attached to them

**Comment by andreasabel** [comment]
Picked for 2.6.2.2-rc2 (with easily fixed merge conflict).
